src/rws_common.cpp

```diff
@@ -201,6 +201,37 @@ bool xmlNodeHasAttribute(const Poco::XML::Node* p_node, const std::string& name,
 }
 
 
+std::string findSubstringContent(const std::string& whole_string,
+                                             const std::string& substring_start,
+                                             const std::string& substring_end)
+{
+  std::string result;
+  size_t start_postion = whole_string.find(substring_start);
+
+  if (start_postion != std::string::npos)
+  {
+    start_postion += substring_start.size();
+    size_t end_postion = whole_string.find_first_of(substring_end, start_postion);
+
+    if (end_postion != std::string::npos)
+    {
+      result = whole_string.substr(start_postion, end_postion - start_postion);
+    }
+  }
+
+  std::string quot = "&quot;";
+  size_t quot_position = 0;
+  do
+  {
+    quot_position = result.find(quot);
+    if (quot_position != std::string::npos)
+    {
+      result.replace(quot_position, quot.size(), "");
+    }
+  } while (quot_position != std::string::npos);
```

Comment:
@Wojcik98 sorry, I could not understand your comment. Could you rephrase it?
This function was just copied "as is" to here from some other part of the code, because it seems to belong here.

Reply:
The function is called `findSubstringContent`, which suggest that it will just return the content found between `substring_start` and `substring_end`. However, this function first finds this content, but then removes from it all substrings `&quot`.
My suggestion was to split it into two functions: first one to just find the content and the other to just remove `&quot` from input string.
But if it was copied over then I guess we can leave it as it is.